api/src/server/mod.rs

```diff
@@ -42,17 +49,36 @@ pub async fn run_server(pool: Pool, host: String) -> io::Result<()> {
     // Init GraphQL schema
     let gql_schema = Arc::new(create_gql_schema());
 
+    let google_client_id = env::var("OPENID_CLIENT_ID").unwrap();
+    let google_client_secret = env::var("OPENID_CLIENT_SECRET").unwrap();
+    let google_url = "https://accounts.google.com";
+    let client =
+        make_client(google_client_id, google_client_secret, google_url).await;
```

Comment:
Do you have a plan for how to handle multiple clients? Should we store them in a map of `name:client`? Also, it'd be nice to move as much of the auth logic out of this file and into `auth.rs` as possible. Maybe once we have multiple clients we could offload all that logic into that file?

Also, does this make a during call during the `.await`? I don't think we should be blocking server start on an outside network call. Is there a way around that?

Reply:
Yea i was thinking a map of name to client. We could probably do all the init stuff in `auth.rs` and just return that map here so this is less bloated.

> Also, does this make a during call during the .await?
Yea, the ` match DiscoveredClient::discover` in `make_client` calls out to a standard url that returns info about how the provider is configured. We might be able to set up a wrapper around the openid client that says if it is ready or not. Then on the first login attempt or w/e we call out to the discover url to configure it. But really it shouldn't take too long on startup to init. Im done for either